gdsfactory/boolean.py

```diff
@@ -70,13 +70,13 @@ def boolean(
     layer_index2 = get_layer(layer2)
     layer_index = get_layer(layer)
 
-    if isinstance(A, Component):
+    if isinstance(A, Component | kf.KCell):
```

Comment:
then just check for KCell, Component inherits from KCell, so checking from KCell is sufficient

Reply:
great point! i fixed it